todo_app/data/trello_repository.py

```diff
@@ -0,0 +1,112 @@
+import requests
+import os
+
+from todo_app.data.models.item import Status
+from todo_app.data.models.trello_objects import TrelloList, TrelloCard
+
+
+class TrelloRepository:
+    def __init__(self):
+        self.headers = {"Accept": "application/json"}
+        self.base_query_params = {
+            "key": os.getenv("TRELLO_API_KEY"),
+            "token": os.getenv("TRELLO_API_TOKEN"),
+            "fields": "id,name,idList,",
+        }
+        self.board_id = os.getenv("TRELLO_BOARD_ID")
+        self.board_url = f"https://api.trello.com/1/boards/{self.board_id}"
+        self.lists_cache = {}
```

Comment:
All of the caching behaviour worked nicely for me, so well done implementing it, but worth also pausing to ask whether it was worthwhile! We could get the current behaviour more simply by just asking users to provide two extra environment variables for the ToDo and Completed list ids respectively.

That would be potentially more flaky than your set up (e.g. if we accidentally put the list ids for board A next to the board id for board B) _but_ it would also remove the amount of code we need to write/maintain substantially, and reduce the amount of work our app is doing so potentially having positive performance implications (albeit your caching will make this pretty minor I think)

For a training exercise, this is definitely better. For a real project, my own instincts would lean towards the simplicity unless we convince ourselves this is necessary!

Reply:
Agreed, I mainly implemented this to see if I could 😅. Also this allows us to add more types of board in future if we want more statuses etc 😄